vstsBuild/vstsBuild.psm1

```diff
@@ -0,0 +1,114 @@
+# VSTS task states: Succeeded|SucceededWithIssues|Failed|Cancelled|Skipped
+$succeededStateName = 'Succeeded'
+$warningStateName = 'SucceededWithIssues'
+$errorStateName = 'Failed'
+
+# store the current state used by *-VstsTaskState and Write-VstsMessage
+$script:taskstate = $succeededStateName
+
+function Clear-VstsTaskState
+{
+    $script:taskstate = $succeededStateName
+}
+
+$script:publishedFiles = @()
```

Comment:
I want to publish the files as soon as possible (after each build), but we should not duplicate publishing any files.  It will make using the artifacts very confusing and make the build slow.